File: backend/app/routes/pnl.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.database import get_db
from app.schemas.pnl import RealizedPnlItem, UnrealizedPnlItem, PnlSummary
from app.services import pnl_service as svc

router = APIRouter(prefix="/api/v1/pnl", tags=["P&L"])


@router.get("/summary", response_model=PnlSummary)
def get_pnl_summary(db: Session = Depends(get_db)):
    return svc.get_pnl_summary(db)


@router.get("/realized", response_model=list[RealizedPnlItem])
def get_realized_pnl(db: Session = Depends(get_db)):
    return svc.get_realized_pnl(db)


@router.get("/unrealized", response_model=list[UnrealizedPnlItem])
def get_unrealized_pnl(db: Session = Depends(get_db)):
    return svc.get_unrealized_pnl(db)

File: frontend/src/api/pnl.ts
import { api } from './client'
import type { PnlSummary, RealizedPnlItem, UnrealizedPnlItem } from '@/types'

const BASE = '/api/v1/pnl'

export const pnlApi = {
  summary: () => api.get<PnlSummary>(`${BASE}/summary`),
  realized: () => api.get<RealizedPnlItem[]>(`${BASE}/realized`),
  unrealized: () => api.get<UnrealizedPnlItem[]>(`${BASE}/unrealized`),
}
